cmd/utils/flags.go

```diff
@@ -2073,7 +2075,7 @@ func MigrateFlags(action func(ctx *cli.Context) error) func(*cli.Context) error
 	return func(ctx *cli.Context) error {
 		for _, name := range ctx.FlagNames() {
 			if ctx.IsSet(name) {
-				ctx.GlobalSet(name, ctx.String(name))
+				ctx.Set(name, ctx.String(name))
```

Comment:
This change turned it into a no-op. You're taking all 'Set' flags and calling 'Set' on them again

Reply:
Hmm, btw, do we still need to do the migration with v2 library? If so what's the main motivation for switching?

Reply:
I thought we wouldn't need to, but turns out we do. As for why switching, I can kind of agree that it's a good idea to move off an abandoned library/version, on to something actively supported.